net-p2p/bitcoin-qt/bitcoin-qt-0.16.0.ebuild

```diff
@@ -0,0 +1,182 @@
+HOMEPAGE="https://bitcoincore.org/ https://bitcoinknots.org/"
+SRC_URI="
+	https://github.com/bitcoin/bitcoin/archive/${BITCOINCORE_COMMITHASH}.tar.gz -> bitcoin-v${PV}.tar.gz
+	https://bitcoinknots.org/files/0.16.x/${KNOTS_PV}/${KNOTS_P}.patches.txz -> ${KNOTS_P}.patches.tar.xz
+"
+
+LICENSE="MIT"
+SLOT="0"
+KEYWORDS="~amd64 ~amd64-linux ~arm ~arm64 ~mips ~ppc ~x86 ~x86-linux"
+
+IUSE="+asm +bip70 +bitcoin_policy_rbf dbus kde +libevent knots libressl +qrcode test upnp +wallet zeromq"
+
+RDEPEND="
+	>=dev-libs/boost-1.52.0:=[threads(+)]
+	>=dev-libs/libsecp256k1-0.0.0_pre20151118:=[recovery]
+	dev-libs/univalue:=
+	dev-qt/qtcore:5
+	dev-qt/qtgui:5
+	dev-qt/qtnetwork:5
+	dev-qt/qtwidgets:5
+	virtual/bitcoin-leveldb
+	bip70? ( dev-libs/protobuf:= )
+	dbus? (
+		dev-qt/qtdbus:5
+	)
+	libevent? ( dev-libs/libevent:= )
+	!libressl? ( dev-libs/openssl:0=[-bindist] )
+	libressl? ( dev-libs/libressl:0= )
+	qrcode? (
+		media-gfx/qrencode:=
+	)
+	upnp? ( >=net-libs/miniupnpc-1.9.20150916:= )
+	wallet? ( sys-libs/db:$(db_ver_to_slot "${DB_VER}")=[cxx] )
+	zeromq? ( net-libs/zeromq:= )
+"
+DEPEND="${RDEPEND}
+	dev-qt/linguist-tools:5
+	knots? (
+		gnome-base/librsvg
+		media-gfx/imagemagick[png]
+	)
+"
+
+DOCS=( doc/bips.md doc/files.md doc/release-notes.md )
+
+S="${WORKDIR}/bitcoin-${BITCOINCORE_COMMITHASH}"
+
+pkg_pretend() {
+	if use knots; then
+		elog "You are building ${PN} from Bitcoin Knots."
+		elog "For more information, see:"
+		elog "https://bitcoinknots.org/files/0.16.x/${KNOTS_PV}/${KNOTS_P}.desc.html"
+	else
+		elog "You are building ${PN} from Bitcoin Core."
+		elog "For more information, see:"
+		elog "https://bitcoincore.org/en/2017/11/11/release-${PV}/"
+	fi
+	if use bitcoin_policy_rbf; then
+		elog "Replace By Fee policy is enabled: Your node will preferentially mine and"
+		elog "relay transactions paying the highest fee, regardless of receive order."
+	else
+		elog "Replace By Fee policy is disabled: Your node will only accept the first"
+		elog "transaction seen consuming a conflicting input, regardless of fee"
+		elog "offered by later ones."
+	fi
+}
+
+src_prepare() {
+	sed -i 's/^\(complete -F _bitcoind \)bitcoind \(bitcoin-qt\)$/\1\2/' contrib/bitcoind.bash-completion || die
+
+	local knots_patchdir="${WORKDIR}/${KNOTS_P}.patches/"
+
+	eapply "${knots_patchdir}/${KNOTS_P}.syslibs.patch"
+
+	if use knots; then
+		eapply "${knots_patchdir}/${KNOTS_P}.f.patch"
+		eapply "${knots_patchdir}/${KNOTS_P}.branding.patch"
+		eapply "${knots_patchdir}/${KNOTS_P}.ts.patch"
+		eapply "${FILESDIR}/${P}-fix_mempoolstats.patch"
+	fi
+
+	eapply_user
+
+	if ! use bitcoin_policy_rbf; then
+		sed -i 's/\(DEFAULT_ENABLE_REPLACEMENT = \)true/\1false/' src/validation.h || die
+	fi
+
+	echo '#!/bin/true' >share/genbuild.sh || die
+	mkdir -p src/obj || die
+	echo "#define BUILD_SUFFIX gentoo${PVR#${PV}}" >src/obj/build.h || die
+
+	sed -i 's/^\(Icon=\).*$/\1bitcoin-qt/;s/^\(Categories=.*\)$/\1P2P;Network;Qt;/' contrib/debian/bitcoin-qt.desktop || die
```

Comment:
Sorry, I just noticed this now. But why are you replacing Icon=bitcoin128 with Icon=bitcoin-qt. Both bitcoin and breeze theme ship icon called bitcoin128

Reply:
The correct icon is installed as `/usr/share/pixmaps/bitcoin-qt.ico`

`bitcoin128.png` only has a single resolution.

I suppose I should see about getting this upstreamed.

Reply:
Somehow my plasma desktop fails to pick the icon from /usr/share/pixmaps/

I am not sure why. Could it be because it's .ico?

Reply:
Seems to be. Per our conversation on IRC, I'll PR a revbump shortly.

Reply:
Oh yes, fd.o specs don't support `.ico` format.